IntegrationTests/Services/AWSSTSIntegrationTests/DefaultAWSCredentialIdentityResolverChainTests.swift

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import AWSSTS
+import ClientRuntime
+import class SmithyHTTPAPI.HTTPRequest
+import class SmithyHTTPAPI.HTTPResponse
+import enum Smithy.ClientError
+import struct SmithyIdentity.AWSCredentialIdentity
+
+class DefaultAWSCredentialIdentityResolverChainTests: XCTestCase {
+    private var stsClient: STSClient!
+    private let region = "us-west-2"
+
+    override func setUp() async throws {
+        try await super.setUp()
+        let stsConfig = try await STSClient.STSClientConfiguration(region: region)
+        stsConfig.addInterceptorProvider(XCTestAssertionInterceptorProvider())
+        stsClient = STSClient(config: stsConfig)
+    }
+
+    func testCredentialsCaching() async throws {
+        let input = GetCallerIdentityInput()
+        for _ in 1...5 {
+            _ = try await stsClient.getCallerIdentity(input: input)
+        }
+    }
+
+    fileprivate class XCTestAssertionInterceptor<InputType, OutputType>: Interceptor {
```

Comment:
I like this use of interceptors for test purposes.

Does the test fail when the caching is removed from the default chain?

Reply:
In GitHub CI it passes regardless bc we resolve credentials once and save it as environment variables. Regardless, your comment revealed a bug. Every API request gets its own copy of interceptor, so cached credential wasn't actually being compared properly. I changed the interceptor provider to instantiate interceptor once and return that for all subsequent `create` functions and I've confirmed that caching behavior on/off makes test pass/fail when using STS assume role in local.

Reply:
Hmm... the changed integ test doesn't work in iOS and tvOS bc using a parameterized protocol type (i.e., a protocol with associated types) in a type-erased or generic way at runtime isn't supported in older OS versions. Looking for workaround rn.

Reply:
I hadn't thought of the state in the interceptors, glad you found it... a singleton should fix it, good call

Reply:
There's no way to get around the force-cast to a parameterized protocol that causes the compile failure?

Reply:
There might be, but I figured testing it in mac and linux and confirming it behaves as expected (when run in Catapult) is adequate.